src/codegen.cpp

```diff
@@ -2769,10 +2769,8 @@ static bool uses_specsig(jl_value_t *sig, bool needsparams, bool va, jl_value_t
         return false;
     if (jl_nparams(sig) == 0)
         return false;
-    if (va) {
-        if (jl_is_vararg(jl_tparam(sig, jl_nparams(sig) - 1)))
-            return false;
-    }
+    if (jl_vararg_kind(jl_tparam(sig, jl_nparams(sig) - 1)) == JL_VARARG_UNBOUND)
```

Comment:
It doesn't look like the other queries in this function will be computed correctly for a bound or numeric var (being based on nparams instead of nargs+isva). Though that seems a somewhat pre-existing problem.

Reply:
Yes, we can adjust that, but I don't think we ever did the change to actually normalize it bound vararg, so I'm not sure it ever actually shows up here at the moment. Regardless, as you said, separate concern, since the reachability of that isn't dependent on the `isva` flag.